File: Classes/OutputClasses/EngineClasses/Nozzle.py
from ...GeneralClasses.ControlSurface import ControlSurface
from ...SensorClasses.PressureSensor import Barometer

class Nozzle(ControlSurface):
    def __init__(self) -> None:
        super().__init__()
        self.min_position.set(0)
        self.change_rate.set(1)
        self.barometer = Barometer()

    def deploy(self):
        if not self.move_to_target():
            return
            #print(f'\n{self.name.get()} is not in tolerance.')

    def retract(self):
        self.return_to_zero()

    def run(self):
        self.calculate_target()
        self.deploy()

    def calculate_target(self):
        delta_p = self.barometer.get()
        tgt_position = self.current_position.get() + delta_p
        self.set_target_position(tgt_position)

File: C++ Implimentation/Classes/Engines/Engine/Nozzle/Nozzle.cpp
#include "Nozzle.h"

    Nozzle::Nozzle()
    : barometer() 
    {
    min_position.setValue(0);
    change_rate.setValue(1);
    }

void Nozzle::calculate_target(){
    int tgt_position = current_position.getValue() + barometer.getValue();
    set_target_position(tgt_position);
}

void Nozzle::deploy(){
    if (move_to_target().getNotValue()){
        return;
        //#print(f'\n{self.name.get()} is not in tolerance.')
        }
}

void Nozzle::retract(){
    return_to_zero();
}

void Nozzle::run(){
    calculate_target();
    deploy();
}
